compiler/src/dotty/tools/dotc/transform/PostTyper.scala

```diff
@@ -147,6 +147,14 @@ class PostTyper extends MacroTransform with IdentityDenotTransformer { thisPhase
         case tree: ValOrDefDef if !tree.symbol.is(Synthetic) =>
           checkInferredWellFormed(tree.tpt)
           val sym = tree.symbol
+          if sym.is(Method) then
+            if sym.isSetter then
+              removeUnwantedAnnotations(sym, defn.SetterMetaAnnot, NoSymbol, keepIfNoRelevantAnnot = false)
+          else
+            if sym.is(Param) then
+              removeUnwantedAnnotations(sym, defn.ParamMetaAnnot, NoSymbol, keepIfNoRelevantAnnot = true)
+            else
+              removeUnwantedAnnotations(sym, defn.GetterMetaAnnot, defn.FieldMetaAnnot, keepIfNoRelevantAnnot = !sym.is(ParamAccessor))
```

Comment:
Thanks for implementing this stuff! This should fix https://github.com/lampepfl/dotty/issues/9881

Reply:
Ah well, I didn't really have a choice. I had tests that needed to pass. 🙃

Reply:
Can you add the testcase from that issue to make sure it's actually fixed?

Reply:
Done.